docs/painless/painless-operators-boolean.asciidoc

```diff
@@ -0,0 +1,516 @@
+[[painless-operators-boolean]]
+=== Operators: Boolean
+
+[[boolean-not-operator]]
+==== Boolean Not
+
+Boolean not will flip a boolean value from true to false or false to true using the bang operator.  The format is a bang operator followed by an expression.
+
+*Grammar:*
+[source,ANTLR4]
+----
+boolean_not: '!' expression;
+----
+
+Note that def types will be assumed to be of the boolean type.  Any def type evaluated at run-time that does not represent a boolean will result in an error.  Non-boolean expressions will result in an error.
+
```

Comment:
Generally avoid future tense. Also, runtime doesn't need to be hyphenated. I'd say "Note that def types are assumed to be of type boolean. Any def type evaluated at runtime that does not represent a boolean results in an error."

The last sentence is not clear--is that, "Preceding a non-boolean expression with the  bang operator results in an error."?

Reply:
Changed.